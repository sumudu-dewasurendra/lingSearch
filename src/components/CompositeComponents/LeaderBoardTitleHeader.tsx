import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import Button from '../Button';
import {COLORS} from '../../theme/colors';

const styles = StyleSheet.create({
  mainWrapper: {
    width: '100%',
    height: 65,
    alignItems: 'center',
    justifyContent: 'space-between',
    flexDirection: 'row',
  },
  rankContainer: {
    width: 80,
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 5,
  },
  nameContainer: {
    width: '60%',
    height: 60,
    justifyContent: 'center',
    alignItems: 'flex-start',
    padding: 5,
  },
  bananasContainer: {
    width: '40%',
    height: 60,
    justifyContent: 'center',
    alignItems: 'flex-end',
    padding: 5,
  },
  rightInnerContainer: {
    width: '80%',
    height: 60,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    fontWeight: 'bold',
    fontSize: 15,
  },
  button: {
    padding: 0,
    borderWidth: 0,
  },
});

type LeaderBoardTitleHeaderProps = {
  onPressName: () => void;
  onPressRank: () => void;
};

const LeaderBoardTitleHeader = ({
  onPressName,
  onPressRank,
}: LeaderBoardTitleHeaderProps) => {
  return (
    <View style={styles.mainWrapper}>
      <View style={styles.rankContainer}>
        <Button
          label={'Rank 🔃'}
          labelColor={COLORS.TEXT_COLOR_BLACK}
          onPressButton={onPressRank}
          wrapperStyles={styles.button}
        />
      </View>
      <View style={styles.rightInnerContainer}>
        <View style={styles.nameContainer}>
          <Button
            label={'Name 🔃'}
            labelColor={COLORS.TEXT_COLOR_BLACK}
            onPressButton={onPressName}
            wrapperStyles={styles.button}
          />
        </View>
        <View style={styles.bananasContainer}>
          <Text style={styles.text}>Bananas🍌</Text>
        </View>
      </View>
    </View>
  );
};

export default LeaderBoardTitleHeader;
