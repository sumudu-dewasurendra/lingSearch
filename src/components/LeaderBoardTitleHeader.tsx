import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

const styles = StyleSheet.create({
  mainWrapper: {
    width: '100%',
    height: 65,
    alignItems: 'center',
    justifyContent: 'space-between',
    flexDirection: 'row',
  },
  rankContainer: {
    width: 60,
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 5,
  },
  nameContainer: {
    width: '60%',
    height: 60,
    justifyContent: 'center',
    alignItems: 'flex-start',
    padding: 5,
  },
  bananasContainer: {
    width: '40%',
    height: 60,
    justifyContent: 'center',
    alignItems: 'flex-end',
    padding: 5,
  },
  rightInnerContainer: {
    width: '80%',
    height: 60,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    fontWeight: 'bold',
    fontSize: 15,
  },
});
const LeaderBoardTitleHeader = () => {
  return (
    <View style={styles.mainWrapper}>
      <View style={styles.rankContainer}>
        <Text style={styles.text}>Rank</Text>
      </View>
      <View style={styles.rightInnerContainer}>
        <View style={styles.nameContainer}>
          <Text style={styles.text}>Name</Text>
        </View>
        <View style={styles.bananasContainer}>
          <Text style={styles.text}>Bananas🍌</Text>
        </View>
      </View>
    </View>
  );
};

export default LeaderBoardTitleHeader;
