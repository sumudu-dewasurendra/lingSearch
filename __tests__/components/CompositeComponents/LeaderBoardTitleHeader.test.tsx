import React from 'react';
import {create} from 'react-test-renderer';
import LeaderBoardTitleHeader from '../../../src/components/CompositeComponents/LeaderBoardTitleHeader';
import {expect, test} from '@jest/globals';

test('renders LeaderBoardTitleHeader component correctly', () => {
  const tree = create(
    <LeaderBoardTitleHeader
      sortAlphabetically={true}
      onPressName={() => console.log('onPressName')}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
