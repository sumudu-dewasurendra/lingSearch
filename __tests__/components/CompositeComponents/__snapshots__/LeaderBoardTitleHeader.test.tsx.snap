// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LeaderBoardTitleHeader component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 65,
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "padding": 5,
        "width": 80,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#c5c5af",
          "borderRadius": undefined,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 0,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 15,
            "fontWeight": "700",
            "letterSpacing": 0.5,
          }
        }
      >
        Rank 🔃
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "width": "80%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "padding": 5,
          "width": "60%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#c5c5af",
            "borderRadius": undefined,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": undefined,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 0,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000000",
              "fontSize": 15,
              "fontWeight": "700",
              "letterSpacing": 0.5,
            }
          }
        >
          Name 🔃
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "height": 60,
          "justifyContent": "center",
          "padding": 5,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        Bananas🍌
      </Text>
    </View>
  </View>
</View>
`;
